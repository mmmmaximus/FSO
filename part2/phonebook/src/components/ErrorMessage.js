import React from 'react'

const ErrorMessage = ({message, type}) => {
  if (message === '') {
    return null
  }
  if (type === 'error') {
    return <div data-testid='ErrorMessage' className='error'>{message}</div>
  } else {
    return <div data-testid='ErrorMessage' className='notification'>{message}</div>
  }
}

export default ErrorMessage
