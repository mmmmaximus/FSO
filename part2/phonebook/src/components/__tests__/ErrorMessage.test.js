import { render, screen } from '@testing-library/react'
import ErrorMessage from '../ErrorMessage'
import renderer from 'react-test-renderer'

describe('ErrorMessage component renders properly when', () => {
  it('has error message', () => {
    render(<ErrorMessage message='error text' type='error'/>)
    const ErrorMessageElement = screen.getByTestId('ErrorMessage')
    expect(ErrorMessageElement.textContent).toBe('error text')
  })

  it('has notification message', () => {
    render(<ErrorMessage message='notification text' type='error'/>)
    const ErrorMessageElement = screen.getByTestId('ErrorMessage')
    expect(ErrorMessageElement.textContent).toBe('notification text')
  })

  it('matches snapshot', () => {
    const tree = renderer.create(<ErrorMessage message='error message' type=''/>).toJSON()
    expect(tree).toMatchSnapshot()
  })
})
