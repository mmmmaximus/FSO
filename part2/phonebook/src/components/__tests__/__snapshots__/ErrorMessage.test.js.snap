// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ErrorMessage component renders properly when matches snapshot 1`] = `
<div
  className="notification"
  data-testid="ErrorMessage"
>
  error message
</div>
`;
